honeycast/message_channel.py

```diff
@@ -0,0 +1,74 @@
+import socket
+import ssl
+import socketserver
+import logging
+from http.server import BaseHTTPRequestHandler
+from struct import pack, unpack
+from multiprocessing import Process
+
+import pdb
+
+import cast_channel_pb2
+
+logger = logging.getLogger("honeycast-message-channel")
+
+def recvall(sock, n):
+    data = b''
+    while len(data) < n:
+        packet = sock.recv(n - len(data))
+        if not packet:
+            return None
+        data += packet
+    return data
+
+class MessageChannelSSLServer(socketserver.TCPServer):
+    def get_request(self):
+        sock, from_addr = self.socket.accept()
+        ssl_sock = ssl.wrap_socket(sock, keyfile="key.pem",
+                certfile="cert.pem", server_side=True)
+        return ssl_sock, from_addr
+
+class MessageChannelHandler(BaseHTTPRequestHandler):
+    def handle(self):
+        sock = self.request
+        packet_size, = unpack(">I", recvall(sock, 4))
+
+        data = recvall(sock, packet_size)
+
+        cast_message = cast_channel_pb2.CastMessage()
+        cast_message.ParseFromString(data)
+
+        device_auth_message = cast_channel_pb2.DeviceAuthMessage()
+        device_auth_message.ParseFromString(cast_message.payload_binary)
+
+        auth_response = cast_channel_pb2.AuthResponse()
```

Comment:
If we are able to skip the authorization. At this point, we will be able to intercept the messages and send OK responses. These messages control what the cast should do so we would put them all in a log file.